pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdConnectors.java

```diff
@@ -0,0 +1,601 @@
+public class CmdConnectors extends CmdBase {
+
+    private final CreateSource createSource;
+    private final CreateSink createSink;
+    private final DeleteConnector deleteConnector;
+    private final LocalSourceRunner localSourceRunner;
+    private final LocalSinkRunner localSinkRunner;
+
+    public CmdConnectors(PulsarAdmin admin) {
+        super("connectors", admin);
+        createSource = new CreateSource();
+        createSink = new CreateSink();
+        deleteConnector = new DeleteConnector();
+        localSourceRunner = new LocalSourceRunner();
+        localSinkRunner = new LocalSinkRunner();
+
+        jcommander.addCommand("create-source", createSource);
+        jcommander.addCommand("create-sink", createSink);
+        jcommander.addCommand("delete", deleteConnector);
+        jcommander.addCommand("localrun-source", localSourceRunner);
+        jcommander.addCommand("localrun-sink", localSinkRunner);
+    }
+
+    /**
+     * Base command
+     */
+    @Getter
+    abstract class BaseCommand extends CliCommand {
+        @Override
+        void run() throws Exception {
+            processArguments();
+            runCmd();
+        }
+
+        void processArguments() throws Exception {
+        }
+
+        abstract void runCmd() throws Exception;
+    }
+
+    @Parameters(commandDescription = "Run the Pulsar Function locally (rather than deploying it to the Pulsar cluster)")
+
+    class LocalSourceRunner extends CreateSource {
+
+        @Parameter(names = "--brokerServiceUrl", description = "The URL for the Pulsar broker")
+        protected String brokerServiceUrl;
+
+        @Override
+        void runCmd() throws Exception {
+            CmdFunctions.startLocalRun(createSourceConfigProto2(sourceConfig),
+                    sourceConfig.getParallelism(), brokerServiceUrl, jarFile, admin);
+        }
+    }
+
+    class LocalSinkRunner extends CreateSink {
+
+        @Parameter(names = "--brokerServiceUrl", description = "The URL for the Pulsar broker")
+        protected String brokerServiceUrl;
+
+        @Override
+        void runCmd() throws Exception {
+            CmdFunctions.startLocalRun(createSinkConfigProto2(sinkConfig),
+                    sinkConfig.getParallelism(), brokerServiceUrl, jarFile, admin);
+        }
+    }
+
+    @Parameters(commandDescription = "Create Pulsar source connectors")
+    class CreateSource extends BaseCommand {
+        @Parameter(names = "--tenant", description = "The function's tenant")
+        protected String tenant;
+        @Parameter(names = "--namespace", description = "The function's namespace")
+        protected String namespace;
+        @Parameter(names = "--name", description = "The function's name")
+        protected String name;
+        @Parameter(names = "--processingGuarantees", description = "The processing guarantees (aka delivery semantics) applied to the Source")
```

Comment:
should we name it different than processing guarantees since that seems too function specific?

Reply:
messaging guarantees? Do you have a suggestion?

Reply:
transferGuarantees?

Reply:
That is not a known term.  I think people might get confused

Reply:
copySemantics/copyGuarantees?

Reply:
I think we should either call it either messenging guarantees or processing guarantees.  These are known terms.  I would rather not to create our own vernacular for something like this.